Code/GraphMol/Depictor/EmbeddedFrag.cpp

```diff
@@ -483,42 +487,54 @@ static void mirrorTransRingAtoms(const RDKit::ROMol& mol, const RDKit::INT_VECT&
 void EmbeddedFrag::embedFusedRings(const RDKit::VECT_INT_VECT &fusedRings,
                                    bool useRingTemplates) {
   PRECONDITION(dp_mol, "");
-  // ok this is what we are going to do here
-  // embed each of the individual rings. Then
-  // find the largest ring , leave that at the origin
-  // and fuse each of remaining rings
+  // Look for a template for the whole system. Failing that simplify the system
+  // to a set of core atoms and  look for a template for those. If that fails,
+  // start from a single ring. Then add rings one by one
 
-  // get the union of the atoms in the rings
   RDKit::INT_VECT funion;
-  RDKit::Union(fusedRings, funion);
-
+  // look for a template that matches the entire fused ring system
   if (useRingTemplates && fusedRings.size() > 1) {
+    RDKit::Union(fusedRings, funion);
     bool found_template = matchToTemplate(funion, fusedRings.size());
     if (found_template) {
+      // we are done
       return;
     }
   }
-
-  // embed each of the rings independently and find the largest ring
   std::vector<RDGeom::INT_POINT2D_MAP> coords;
   coords.reserve(fusedRings.size());
-  // FIX for issue 197
-  // find the ring with the max substituents
-  // If there are multiple pick the largest
-  auto firstRingId = pickFirstRingToEmbed(*dp_mol, fusedRings);
 
   for (const auto &ring : fusedRings) {
     auto ring_coords = embedRing(ring);
     mirrorTransRingAtoms(*dp_mol, ring, ring_coords);
     coords.push_back(ring_coords);
   }
+  RDKit::INT_VECT doneRings;
 
-  this->initFromRingCoords(fusedRings[firstRingId], coords[firstRingId]);
+  // embed the core rings. Try first with a template if available
+  RDKit::INT_VECT coreRingsIds;
+  auto coreRings = findCoreRings(fusedRings, coreRingsIds);
+  if (useRingTemplates && coreRings.size() > 1) {
+    RDKit::Union(coreRings, funion);
+    bool found_template = matchToTemplate(funion, coreRings.size());
+    if (found_template) {
+      doneRings = coreRingsIds;
+    }
+  }
 
-  RDKit::INT_VECT doneRings;
-  doneRings.push_back(firstRingId);
+  // if not embed find a ring as a starting point
+  if (doneRings.empty()) {
+    // FIX for issue 197
+    // find the ring with the max substituents
+    // If there are multiple pick the largest
+    // auto firstRingId = pickFirstRingToEmbed(*dp_mol, coreRings);
```

Comment:
Not sure if you meant to comment this out

Reply:
yeah, this should go. I was experimenting with choosing a first ring among the core atoms rather than all atoms, but it blows up the tests